src/Microsoft.Health.Fhir.Shared.Api/Features/Filters/OperationOutcomeExceptionFilterAttribute.cs

```diff
@@ -235,6 +247,31 @@ public override void OnActionExecuted(ActionExecutedContext context)
                         context.Exception = outerException;
                     }
                 }
+
+                return;
+            }
+            else
+            {
+                context.Result = new OperationOutcomeResult(
+                    new OperationOutcome
+                    {
+                        Id = _fhirRequestContextAccessor.RequestContext.CorrelationId,
+                        Meta = new Meta()
+                        {
+                            LastUpdated = DateTimeOffset.UtcNow,
+                        },
+                    },
+                    HttpStatusCode.InternalServerError);
+                context.ExceptionHandled = true;
+            }
+
+            if (context.ExceptionHandled)
+            {
+                HttpStatusCode? statusCode = (context.Result as OperationOutcomeResult).StatusCode;
+                if (statusCode != null && statusCode >= HttpStatusCode.InternalServerError)
+                {
+                    _logger.LogError(_fhirRequestContextAccessor.RequestContext.CorrelationId, context.Exception);
```

Comment:
What overload is this? Normally the exception is logged first? i.e. https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.logging.loggerextensions.logerror?view=dotnet-plat-ext-5.0#Microsoft_Extensions_Logging_LoggerExtensions_LogError_Microsoft_Extensions_Logging_ILogger_System_Exception_System_String_System_Object___

Reply:
It is the first one. I'm using the correlation id as an event id.